### ISSUE TITLE
Logging: Track logged messages

### ISSUE THREAD

**Issue by ithinkandicode** [GodotModding/godot-mod-loader#140]
In #138 I added the array variable `logged_messages`. I think this is a potentially very useful approach that could be applied it to all logged messages.

**Implementation:** 

- In `_loader_log`, we save logged messages to an array.
    - This is the same as what I did for #138, 
    - but with the array variable being in `ModLoaderUtils` instead. 
- For each message, we can track all the data parts that are passed as variables to `_loader_log`
  - Ie. mod name, message, and type.
  - Actually, it might be better as a dictionary, with these keys:
      - `all` = All logged messages
      - `by_mod` = Notices logged via a certain mod
      - `by_type` = Notices of a specific type

**API Methods:**

- `get_tracked_messages_all()`
- `get_tracked_messages_by_mod(mod_name: String)`
- `get_tracked_messages_by_type(type: String)` -- where `type` is one of the current log types (`fatal-error`, `error`, etc)

**Use Case:**

This would be a massive help to players who are using mods, as it would save them needing to check the logs (and save them needing to know how to do that). This does depend on the game or a mod presenting this data, but this is the first step towards supporting that.

Note: This isn't related to UI at all. It simply provides data that a UI could interact with, which would be the responsibility of the game developer or modders -- like how Brotato's Mods screen interacts with ModLoader's current data (see example with otDan's mod [BetterModList](https://github.com/otDan/Brotato-BetterModList)). So on that screen, it could include a console-like tray with any errors or warnings.

**Comment by ithinkandicode** [comment]
_Suggestion from @Qubus0 via PR #138 ([here](https://github.com/GodotModding/godot-mod-loader/pull/138/#issuecomment-1446200740)):_

For performance, it could be worth to store all the messages in a dictionary with the keys being their md5 version.
- A it doesn't have to look for the entire message every time
- B dictionary lookup is way faster
- C we can still get all the messages by just using `.values()`

```gdscript
print("Config JSON Notice: something something".md5_text())     # b1f526f21bb565e91f11096f3cc1b499
print("Config JSON Notice: something something".md5_text())     # b1f526f21bb565e91f11096f3cc1b499 - same
print("abc Config JSON Notice: something something".md5_text()) # 34b0bb58ba94c388f7dd361ec0836172 - not same
```

**Comment by ithinkandicode** [comment]
Is this actually possible? The logger funcs are static, so they couldn't save to a `logged_messages` variable in their own class. But they also can't access `ModLoader.logged_messages` either, because ModLoader isn't initialised at the point when the first log method is called.

**Comment by ithinkandicode** [comment]
- Closed by #205
